### PR TITLE
blueprint: fix CA cert testing

### PR DESCRIPTION
**Pull request by lzap**
SSIA @thozza @mvo5

### CODE REVIEW

File: test/scripts/base-host-check.sh [hunk 1]
```diff
@@ -80,16 +80,21 @@ get_oscap_score() {
...
 check_ca_cert() {
+    ls /etc/pki/ca-trust/source/anchors
+
+    jq -r '.blueprint.customizations.cacerts.pem_certs[0]' "${config}"
+    jq -r '.blueprint.customizations.cacerts.pem_certs[0]' "${config}" | openssl x509 -noout -serial
     serial=$(jq -r '.blueprint.customizations.cacerts.pem_certs[0]' "${config}" | openssl x509 -noout -serial | cut -d= -f 2-)
 
+
     echo "📗 Checking CA cert anchor file"
     if ! [ -e "/etc/pki/ca-trust/source/anchors/${serial}.pem" ]; then
```
Comment (thozza) [inline test/scripts/base-host-check.sh:91]: Looking at https://gitlab.com/redhat/services/products/image-builder/ci/images/-/jobs/8586354060, the `${serial}` is uppercase, while the actual filename is lowercase.

File: test/scripts/base-host-check.sh [hunk 2]
```diff
@@ -80,16 +80,16 @@ get_oscap_score() {
 }
 
 check_ca_cert() {
-    serial=$(jq -r '.blueprint.customizations.cacerts.pem_certs[0]' "${config}" | openssl x509 -noout -serial | cut -d= -f 2-)
+    serial=$(jq -r '.blueprint.customizations.cacerts.pem_certs[0]' "${config}" | openssl x509 -noout -serial | cut -d= -f 2- | tr '[:upper:]' '[:lower:]')
 
-    echo "📗 Checking CA cert anchor file"
-    if ! [ -e "/etc/pki/ca-trust/source/anchors/${serial}.pem" ]; then
+    echo "📗 Checking CA cert anchor file serial $serial"
+    if ! [ -e "/etc/pki/ca-trust/source/anchors/27894af897dd2423607045716438a725f28a6d0b.pem" ]; then
```
Comment (thozza) [inline test/scripts/base-host-check.sh:86]: Please use the variable `${serial}` instead.

Comment (lzap) [reply test/scripts/base-host-check.sh]: Done, note the subject is hardcoded few lines below, I could not figure out an easy way of finding the filename, it is generated by `c_rehash` OpenSSL command. A bit more challenging.

Comment (lzap) [reply test/scripts/base-host-check.sh]: I have found a dumb way to generalize this a bit replacing spaces with underscores. To give you some context, the `update-ca-certificates` command copies the certs into the `extracted` directory under `Common_name_with_underscores.pem` and then creates some symlinks. These are not serials, it is something different. The stage does copy the file into `anchors` directory with serial in the name.

**Review by thozza — DISMISSED** [review]
Thanks!

**Comment by lzap** [comment]
Finally it is passing, I figured out how to generalize the test. Ready for review.

File: test/scripts/base-host-check.sh [hunk 3]
```diff
@@ -80,17 +80,20 @@ get_oscap_score() {
...
-    echo "📗 Checking extracted CA cert file"
-    if ! [ -e "/etc/pki/ca-trust/source/extracted/pem/directory-hash/Test_CA_for_osbuild.pem.pem" ]; then
-        echo "Extracted CA file does not exist"
+    echo "📗 Checking extracted CA cert file named '${cn}'"
+    if ! [ -e "/etc/pki/ca-trust/extracted/pem/directory-hash/${cn}.pem" ]; then
+        echo "Extracted CA file does not exist, directory contents:"
+        find /etc/pki/ca-trust/extracted/pem/directory-hash
         exit 1
     fi
 }
```
Comment (lzap) [inline test/scripts/base-host-check.sh:99]: @achilleas-k fyi this test was introduced last week but it was broken (a typo in `directory-hash` path) and due to the config-map it was not executed properly so it sneaked into `main`. This fixes the map and the test.

Comment (achilleas-k) [reply test/scripts/base-host-check.sh:99]: Thank you!

**Review by achilleas-k — APPROVED** [review]

**Comment by lzap** [comment]
Also I have found out that on some (older) systems the directory-hash directory does not exist, so I came with a better test case: it now searches for CN in the bundle PEM file which is concatenated list of all certs with CNs as comments:

```
grep "Test CA for osbuild" /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
```

This should work also on older systems.

**Comment by lzap** [comment]
Is green, let's put it on the queue @achilleas-k @thozza if you like the grep approach.

File: test/scripts/base-host-check.sh [hunk 4]
```diff
@@ -80,17 +80,20 @@ get_oscap_score() {
...
         exit 1
     fi
 
-    echo "📗 Checking extracted CA cert file"
-    if ! [ -e "/etc/pki/ca-trust/source/extracted/pem/directory-hash/Test_CA_for_osbuild.pem.pem" ]; then
-        echo "Extracted CA file does not exist"
+    echo "📗 Checking extracted CA cert file named '${cn}'"
+    if ! grep -q "${cn}" /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem; then
+        echo "Extracted CA file does not exist, directory contents:"
+        find /etc/pki/ca-trust/extracted/pem
```
Comment (thozza) [inline test/scripts/base-host-check.sh:96]: The grep-ing of `tls-ca-bundle.pem` makes me wonder if listing the content of `/etc/pki/ca-trust/extracted/pem` still makes sense? :thinking:

Comment (thozza) [reply test/scripts/base-host-check.sh:96]: So, judging from https://github.com/osbuild/osbuild-composer/pull/4487/files#diff-a0a3d509a4e8903c0bede204d82662223fa270c20360d5bf2dcd2f1d2b68bbcdR264, it should be probably
```shell
grep '^#' /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
```

Comment (thozza) [reply test/scripts/base-host-check.sh]: Thank you for the change. Please also amend the comment above it.

**Comment by lzap** [comment]
Right, done.

**Review by thozza — APPROVED** [review]
Thanks!

**Comment by lzap** [comment]
I do not see any comments by @achilleas-k the tests were fixed, adding to the merge queue. Thanks again for help with this, now I understand how tests do work in here a little bit better :-D

**Pull request merged**